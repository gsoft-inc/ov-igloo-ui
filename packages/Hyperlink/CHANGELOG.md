# @igloo-ui/hyperlink

## 0.1.15

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.

## 0.1.14

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.1.13

### Patch Changes

- d5c91f5: link shared components with aliases

## 0.1.12

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.11

### Patch Changes

- 0640409: Update packages

  - "react-aria": "^3.21.0",
  - "@react-aria/link": "^3.35",
  - "react-stately": "^3.19.0"

- 0640409: udpate classnames to 2.3.2

## 0.1.10

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme

## 0.1.9

### Patch Changes

- cd04a2e: Fix hyperlink secondary + danger color on focus

## 0.1.8

### Patch Changes

- 0a86f13: Update content of README

## 0.1.7

### Patch Changes

- f019f88: fix position of icon with content warp

## 0.1.6

### Patch Changes

- ac5d81a: fix position of icon with content warp

## 0.1.5

### Patch Changes

- ba32c70: fix position of icon with content warp

## 0.1.4

### Patch Changes

- f83c8b4: fix position of icon with content warp

## 0.1.3

### Patch Changes

- 574c7d8: Fix display of translation content

## 0.1.2

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 0.1.1

### Patch Changes

- 4a256e3: IE11 compatibility

## 0.1.0

### Minor Changes

- 2e23826: Initial release

  Component API

  | name           | description                                                                           |
  | -------------- | ------------------------------------------------------------------------------------- |
  | appearance     | link appearance `primary, secondary, premium, ghost, danger`                          |
  | size           | Changes the size of link, giving more or less padding <br/> `small, medium`           |
  | iconLeading    | Icon to display to the left of link content                                           |
  | iconTrailing   | Icon to display to the right of link content                                          |
  | children       | the content to display inside the link                                                |
  | underline      | add a underline on link                                                               |
  | dataTest       | add a data-test tag for automated tests                                               |
  | className      | add a specific class to the link                                                      |
  | intercomTarget | add a data-intercom-target with unique id to link a components to a Product Tour step |
