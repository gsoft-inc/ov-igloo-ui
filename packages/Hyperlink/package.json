{
  "name": "@igloo-ui/hyperlink",
  "version": "0.4.2",
  "main": "dist/Hyperlink.js",
  "module": "dist/Hyperlink.js",
  "types": "dist/Hyperlink.d.ts",
  "styles": "dist/hyperlink.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com",
    "Alexandre alexandre.beaudoin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.1.0",
    "@hopper-ui/tokens": "^3.5.1",
    "@react-aria/link": "^3.6.5",
    "classnames": "^2.3.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
