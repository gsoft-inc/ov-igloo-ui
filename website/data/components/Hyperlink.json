[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Hyperlink/src/Hyperlink.tsx",
      "description": "",
      "displayName": "Hyperlink",
      "methods": [],
      "props": {
        "children": {
          "defaultValue": null,
          "description": "The content to display inside the link",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "size": {
          "defaultValue": null,
          "description": "Changes the size of link, giving more or less padding",
          "name": "size",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "Size" }
        },
        "appearance": {
          "defaultValue": null,
          "description": "Link appearance",
          "name": "appearance",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "Appearance" }
        },
        "underline": {
          "defaultValue": null,
          "description": "Add a underline on link",
          "name": "underline",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "iconLeading": {
          "defaultValue": null,
          "description": "Icon to display to the left of link content",
          "name": "iconLeading",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "iconTrailing": {
          "defaultValue": null,
          "description": "Icon to display to the right of link content",
          "name": "iconTrailing",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the link",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "intercomTarget": {
          "defaultValue": null,
          "description": "Add a data-intercom-target with unique id\nto link a components to a Product Tour step",
          "name": "intercomTarget",
          "parent": {
            "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
            "name": "HyperlinkProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Hyperlink/src/Hyperlink.tsx",
              "name": "HyperlinkProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
