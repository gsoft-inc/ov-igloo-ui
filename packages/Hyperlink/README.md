# Hyperlink

Hyperlink are interactive elements that allow users to navigate in webpage.

<ReferenceLinks />

## Installation

To install `@igloo-ui/hyperlink` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/hyperlink
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/hyperlink
```

Then to use the component in your code just import it!

```jsx
import Hyperlink from '@igloo-ui/hyperlink';

<Hyperlink>
  <a href="#">Link to ...</a>
</Hyperlink>;
```
