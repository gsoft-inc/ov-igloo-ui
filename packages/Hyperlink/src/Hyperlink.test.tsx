/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import Plus from '@igloo-ui/icons/dist/Plus';

import Hyperlink from './Hyperlink';

const setUp = (props = {}) => {
  const component = shallow(
    <Hyperlink dataTest="ids-link" {...props}>
      Click me
    </Hyperlink>
  );
  return component;
};

describe('Hyperlink Component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let component: any;
  beforeEach(() => {
    component = setUp();
  });

  test('It should render without errors', () => {
    const wrapper = component.find('.ids-link');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a hyperlink with icon', (): void => {
    render(<Hyperlink iconLeading={<Plus size="small" />}>Click me</Hyperlink>);
    const svg = document.getElementsByTagName('svg');
    expect(svg.length).toBe(1);
  });

  test('It should render a primary appearance', (): void => {
    const hyperlink = setUp();
    expect(hyperlink).toMatchSnapshot();
  });

  test('It should render a secondary appearance', (): void => {
    const hyperlink = setUp({ appearance: 'secondary' });
    expect(hyperlink).toMatchSnapshot();
  });

  test('It should render a premium appearance', (): void => {
    const hyperlink = setUp({ appearance: 'premium' });
    expect(hyperlink).toMatchSnapshot();
  });

  test('It should render a danger appearance', (): void => {
    const hyperlink = setUp({ appearance: 'danger' });
    expect(hyperlink).toMatchSnapshot();
  });

  test('It should render a ghost appearance', (): void => {
    const hyperlink = setUp({ appearance: 'ghost' });
    expect(hyperlink).toMatchSnapshot();
  });
});
