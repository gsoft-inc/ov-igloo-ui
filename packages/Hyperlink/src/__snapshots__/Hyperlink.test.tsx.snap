// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyperlink Component It should render a danger appearance 1`] = `ShallowWrapper {}`;

exports[`Hyperlink Component It should render a ghost appearance 1`] = `ShallowWrapper {}`;

exports[`Hyperlink Component It should render a premium appearance 1`] = `ShallowWrapper {}`;

exports[`Hyperlink Component It should render a primary appearance 1`] = `ShallowWrapper {}`;

exports[`Hyperlink Component It should render a secondary appearance 1`] = `ShallowWrapper {}`;
